src/state.ts

```diff
@@ -136,6 +130,15 @@ notBackground &&
         if (message.command == "stateUpdate") {
             const property = message.args.property
             const value = message.args.value
+            // Ensure we don't accidentally store anything sensitive
+            const inIncognitoContext = message.args.inIncognitoContext
+            if (inIncognitoContext && PERSISTENT_KEYS.includes(property)) {
+                overlay[property] = value
```

Comment:
What is this line for?

(Thanks for looking at this, by the way!)

Reply:
I think I'm going to remove this line - otherwise you could

1. set the property to something sensitive
2. reject the write
3. set another property to something not at all sensitive
4. write it all including the sensitive property

Please let me know if I'm missing something : )